discovery-provider/src/queries/query_helpers.py

```diff
@@ -328,6 +329,85 @@ def populate_track_metadata(session, track_ids, tracks, current_user_id):
     )
     save_count_dict = {track_id: save_count for (track_id, save_count) in save_counts}
 
+    # Fetch the remix parent track's user and if that user has saved/favorited the child track 
```

Comment:
Nit: populate_track_metadata is getting pretty massive now, I think we should consider pulling all out all of this remix log into a helper that we call from this function